### PR TITLE
#160451571 users should be able to view notifications

### PR DESCRIPTION
**Pull request by e0works**
#### Description of Task to be completed?
- write action tests for notifications
- get user unread notifications when authenticated
- mark a single notification as read
- mark all notifications as read

#### How should this be manually tested?
- start the app
- you should see all your notification after you are authenticated
- click on each notification item to clear or
- click on `mark all as read` to clear all

#### What are the relevant pivotal tracker stories?
[#160451571](https://www.pivotaltracker.com/story/show/160451571, "view user notification")

#### Screenshots (if appropriate)
screen shot 2018-11-16 at 3 16 04 pm

### CODE REVIEW

File: src/action/notification/__test__/getNotification.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+// actionType
+import {
+  VIEW_NOTIFICATION,
+  VIEW_NOTIFICATION_SUCCESS,
+  VIEW_NOTIFICATION_FAILURE,
+} from '../../../actionTypes/notification';
+
+
+
+// action
```
Comment (KaiserPhemi) [inline src/action/notification/__test__/getNotification.js:14]: #### Nitpick
There should only be one blank line between this line and previous one above it

File: src/components/fixtures/notification/NotificationAction.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// react libraries
+import React from 'react';
+
+
```
Comment (KaiserPhemi) [inline src/components/fixtures/notification/NotificationAction.jsx]: #### Nitpick
What are you doing with all these blank lines?

File: src/components/fixtures/notification/NotificationAction.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+// third-party libraries
+import PropTypes from 'prop-types';
+
+//fixtures import
+import markNotificationAsReadAction from '../../../action/notification/readNotification'; 
+import configureStore from '../../../store/configureStore';
+
+// action
```
Comment (KaiserPhemi) [inline src/components/fixtures/notification/NotificationAction.jsx]: #### Nitpick
What's this comment solving for?

File: src/components/reusables/dropdown/notificationDropdown/_NotificationDropdown.scss [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+      color: $primary;
+    }
+  }
+  .no-notification {
+    height: 100px;
+    p {
+      padding-top: 30px;
+    }
+  }
+}
```
Comment (KaiserPhemi) [inline src/components/reusables/dropdown/notificationDropdown/_NotificationDropdown.scss]: #### Nitpick
Add a new line here.

File: src/containers/article/CreateArticle.jsx [hunk 5]
```diff
@@ -8,7 +8,9 @@ import CreateArticlePage from '../../components/article/CreateArticlePage';
 import createNewArticle from '../../action/article/createNewArticle';
 
 const mapStateToProps = (state = {}) => ({
```
Comment (KaiserPhemi) [inline src/containers/article/CreateArticle.jsx]: #### Critical
Why are you passing an empty state here?

File: src/containers/LandingPage/LandingPage.jsx [hunk 6]
```diff
@@ -4,9 +4,15 @@ import { connect } from 'react-redux';
...
+import markNotificationAsReadAction from '../../action/notification/readNotification';
 
 const mapStateToProps = state => ({
-  homeLogin: state.auth
+  homeLogin: state.auth,
+  notifications: state.getNotification,
 });
 
-export default connect(mapStateToProps)(LandingPage);
+const mapDispatchToProps = dispatch => ({
```
Comment (KaiserPhemi) [inline src/containers/LandingPage/LandingPage.jsx]: #### Critical
You can refactor this. i.e.
```js
const mapDispatchToProps = {
  markNotificationAsRead(token, id, mark) { markNotificationAsReadAction(token, id, mark); }
}
```
You can check [here](https://daveceddia.com/redux-mapdispatchtoprops-object-form/) for clarity

File: src/components/reusables/dropdown/notificationDropdown/NotificationDropdown.jsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
+// react libraries
+import React, { Component } from 'react';
+
+// third party libraries
+import PropTypes from 'prop-types';
+import NotificationItem from '../../notification/NotificationItem';
```
Comment (KaiserPhemi) [inline src/components/reusables/dropdown/notificationDropdown/NotificationDropdown.jsx:8]: #### Critical
This should have it's own comment 
i.e. 
```js
// component
```

File: src/components/reusables/dropdown/menuDropdown/Dropdown.jsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import { Link } from 'react-router-dom';
+import propTypes from 'prop-types';
+
+/**
+ * @returns {object} element
+ * @param {object} props
+ */
+export default function Dropdown(props) {
+  const { active } = props;
+  const menuArray = [
```
Comment (KaiserPhemi) [inline src/components/reusables/dropdown/menuDropdown/Dropdown.jsx]: #### Critical
Drop this into a fixture file and call it within the component.

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]

**Review by obulaworld — APPROVED** [review]
Looking Good to Me

**Review by andela-phemmy — APPROVED** [review]
Implement the feedbacks. Looking Good to Me

File: src/containers/LandingPage/LandingPage.jsx (hunk 6 above)
Comment (e0works) [reply src/containers/LandingPage/LandingPage.jsx]: I have gone through the article, this what I saw as a sample code
```js
const mapDispatchToProps = {
  decrement,
  increment: () => increment(42),
  reset
};
```

If I use it this way,

## 1
```js
const mapDispatchToProps = {
  markNotificationAsRead(token, id, mark) { markNotificationAsReadAction(token, id, mark); }
}
````
it throws error

So, I have to write this way for it to work 
## 2
```js
const mapDispatchToProps = {
  markNotificationAsRead: (token, id, mark) => markNotificationAsReadAction(token, id, mark) 
}
````
I haven't been able to make it work number one

File: src/components/reusables/dropdown/notificationDropdown/NotificationDropdown.jsx (hunk 7 above)
Comment (e0works) [reply src/components/reusables/dropdown/notificationDropdown/NotificationDropdown.jsx:8]: I have implemented this feedback

**Review by KaiserPhemi — APPROVED** [review]

**Pull request merged**